design.md

```diff
@@ -0,0 +1,106 @@
+# oras multi-arch feature design, Version July 15
+
+## PoC
+
+https://github.com/wangxiaoxuan273/oras/pull/1
+
+Only implemented `oras manifest index create`.
+
+## Design
+### All commands
+`oras manifest index create` Create an image index from source manifests. 
+
+`oras manifest index update` Add/Remove manifests to/from an image index.
+
+### oras manifest index create
+
+Create an image index from source manifests. The command auto-detects platform information for each source manifests. The sources and the target index must be in the same repository.
+
+#### Usage
+
+`oras manifest index create [flags] <name>[:<tag>|@<digest>] [{<tag>|<digest>}...]`
+
+#### Flags
+
+(These flags are not implemented in the PoC)
+
+`--subject` Add a subject manifest for the to-be-created index.
+
+`--oci-layout` Set the given repository as an OCI layout.
+
+`--annotation` Add annotations for the to-be-created index.
+
+`--artifact-type` Add artifact type information for the to-be-created index.
```

Comment:
Also need to add `--tag` here.

Reply:
We don't need `--tag` for `oras manifest index create`, as users can put tags after the repo reference, i.e. `localhost:5000/test:latest,v1`

Reply:
Yes, it's doable but depending on how we define first argument. 

For example, if it's strictly defined as the repository, `oras manifest index create localhost:5000/test --tag latest,v1` is a better choice. Also, using `--tag` is more consistent with the UX of`oras manifest index update`.